file-events/src/file-events/cpp/linux_fsnotifier.cpp

```diff
@@ -348,15 +354,17 @@ void Server::stopWatchingMovedPaths(const vector<u16string>& absolutePathsToChec
             continue;
         }
 
-        jstring javaPath = env->NewString((jchar*) watchPoint.path.c_str(), (jsize) watchPoint.path.length());
-        env->CallBooleanMethod(droppedPaths, listAddMethod, javaPath);
-        env->DeleteLocalRef(javaPath);
-        getJavaExceptionAndPrintStacktrace(env);
+        addToList(env, droppedPaths, jPathToCheck);
 
         watchPoint.cancel();
     }
 }
 
+void Server::addToList(JNIEnv* env, jobject jList, jstring jString) {
+        env->CallBooleanMethod(jList, listAddMethod, jString);
+        getJavaExceptionAndPrintStacktrace(env);
```

Comment:
Why do we get exception and print stacktrace here?

Reply:
I suppose we don't want to fail for some reason. This was like that before. Do you know why?

Reply:
Ah, I copied it over from somewhere else. We should remove this.

Reply:
I think we just remove this call. A Java exception is just a thread-local, basically. It will be passed to the unexpected exception handler associated with the thread if one occurs when we return from the JNI call anyway.

Reply:
Ok, I removed the logging of the stacktrace.

Reply:
Aha, when I remove this I get:

```
WARNING in native method: JNI call made without checking exceptions when required to from CallBooleanMethodV
```

What is the right approach here?

Reply:
I think now I got it right: I am throwing now `JavaExceptionThrownException` and catch it in the JNI method to have the right control flow.

Reply:
I moved back to the old handling.

Reply:
We talked again, and since we need to do something with the exception because of the JNI checks, we decided that throwing it is the right thing to do.